tests/io/qasm_interpreter/test_interpreter.py

```diff
@@ -199,22 +206,31 @@ def test_parses_simple_qasm(self, mocker):
         ry = mocker.spy(RY, "__init__")
 
         # execute the callable
-        context["callable"]()
-
-        # asserts
-        assert len(context["wires"]) == 2
+        with queuing.AnnotatedQueue() as q:
+            context["callable"]()
+
+            # asserts
+            assert len(context["wires"]) == 2
+
+            assert x.call_count == 5  # RX calls PauliX under the hood
+            assert cx.call_count == 2  # verifies that ctrl @ x q1, q0 calls cx too
+            assert rx.call_count == 2  # one adjoint call and one direct call
+            assert ry.call_count == 1
+
+            assert q.queue == [
+                PauliX(0),
+                CNOT(wires=[0, 1]),
+                RX(0.5, wires=[0]),
+                RY(0.2, wires=[0]),
+                Adjoint(RX(0.5, wires=[0])),
+                PowOpObs(PauliX(wires=[0]), 2),
+                CNOT(wires=[1, 0]),
+            ]
```

Comment:
Typically we retrieve the queue outside the context.
```suggestion
        assert q.queue == [
            PauliX(0),
            CNOT(wires=[0, 1]),
            RX(0.5, wires=[0]),
            RY(0.2, wires=[0]),
            Adjoint(RX(0.5, wires=[0])),
            PowOpObs(PauliX(wires=[0]), 2),
            CNOT(wires=[1, 0]),
        ]
```

Reply:
Good to understand!